lib/src/web_socket_handler.dart

```diff
@@ -78,26 +78,26 @@ class WebSocketHandler {
       if (protocol != null) sink.add('Sec-WebSocket-Protocol: $protocol\r\n');
       sink.add('\r\n');
 
-      _onConnection(
+      _onConnection?.call(
           WebSocketChannel(channel, pingInterval: _pingInterval), protocol);
     });
 
     // [request.hijack] is guaranteed to throw a [HijackException], so we'll
     // never get here.
     assert(false);
-    return null;
+    throw const HijackException();
```

Comment:
I think this is really a state error? `throw StateError('unreachable');`

However if `request.hijack` always throws it should have a return type of `Never` which should mean you don't have to return anything here.

Reply:
done

Reply:
We should consider fixing `shelf` so that `Request.hijack` has a return type of never before making this change - cc @natebosch .

Reply:
I am currently trying to check if that change would be breaking for us internally, assuming I can fix things up reasonably there we can go ahead and make the change.

Reply:
Actually, we can go ahead and land this as is today now that I think about it. We can update this later.